### PR TITLE
Kimchanyoung requests resources

### PR DESCRIPTION
**Pull request by kimchanyoung**
@StevenXL @saraho-shea 
Oops forgot to add presence validations to association validations

### CODE REVIEW

**Comment by kimchanyoung** [comment]
@StevenXL @saraho-shea 
Features added:
- Users can make requests
- Users can cancel requests
- Users can respond to requests to become responders

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @request = Request.new
+  end
+
+  def create
+    @request = Request.new(request_attributes)
+    @request.requester = current_user
+    if @request.save
+      redirect_to request(@request)
+    else
+      flash.new[:alert] = @request.errors.full_messages.join(", ")
```
Comment (StevenXL) [inline app/controllers/requests_controller.rb]: I think this should be `flash.now`.

**Comment by kimchanyoung** [comment]
@StevenXL Fixed!

File: app/controllers/requests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      render 'new'
+    end
+  end
+
+  def show
+    @request = Request.find_by(params[:id])
```
Comment (StevenXL) [inline app/controllers/requests_controller.rb]: Does this work? We might have to use `@request = Request.find_by(id: params[:id])`. 

I see this on a few different lines so I won't comment on all of them.

File: app/controllers/requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  def update
+    request = Request.find_by(params[:id])
+    request.responder = current_user if request.requester != current_user
```
Comment (StevenXL) [inline app/controllers/requests_controller.rb:29]: Nice.

**Comment by kimchanyoung** [comment]
@StevenXL Thanks for catching that -- I always make this mistake. Fxied!

File: app/controllers/requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    redirect_to request(@request)
+  end
+
+  def destroy
+    request = Request.find_by(params[:id])
+    request.destroy if request.requester == current user
```
Comment (StevenXL) [inline app/controllers/requests_controller.rb:35]: I think we need a bit more logic in here. We should include a flash message if the request was successfull. We should redirect to somewhere if the request is successful. 

We should include a flash.now if the request is not successful, and we should render that same request. 

We can re-direct to root no matter what if you like, but we would still need a flash message.

File: app/controllers/requests_controller.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    redirect_to root
```
Comment (StevenXL) [inline app/controllers/requests_controller.rb]: This will throw an error unless we do `root_path` or URL, I like path..

**Comment by StevenXL** [comment]
:+1:

**Pull request merged**